### PR TITLE
feat(CB2-6091): enable amend certificate and anniversary date

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## CB2-6091: Amend Certificate Expiry and Anniversary Date field

_PR to make the Expiry Date and Certificate Anniversary fields editable when amending a test result_
[CB2-6091](https://dvsa.atlassian.net/browse/CB2-6091)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/forms/validators/custom-validators.ts [hunk 1]
```diff
@@ -179,6 +179,22 @@ export class CustomValidators {
...
   };
 
+  static dateNotExceed = (sibling: string, months: number): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null => {
+      if (control?.parent) {
+        const siblingControl = control.parent.get(sibling) as CustomFormControl;
+        const siblingDate = new Date(siblingControl.value);
+        const maxDate = new Date(siblingDate.setMonth(siblingDate.getMonth() + months));
+
+        if (new Date(control.value) > maxDate) {
```
Comment (gjulien-bjss) [inline src/app/forms/validators/custom-validators.ts:189]:
```suggestion
       siblingDate.setMonth(siblingDate.getMonth() + months);

        if (new Date(control.value) > siblingDate) {
```

**Review by mcadenas-bjss — APPROVED** [review]

**Review by gjulien-bjss — COMMENTED** [review]
I'd say worth checking if those are the only tests this field should be editable on, since the ticket hasn't really done trough a proper refinement

**Comment by leifkemp-bjss** [comment]
> I'd say worth checking if those are the only tests this field should be editable on, since the ticket hasn't really done trough a proper refinement

DM'd Matt about it, he'll get a list written up

File: src/app/forms/validators/custom-validators.ts [hunk 2]
```diff
@@ -179,6 +179,22 @@ export class CustomValidators {
...
   };
 
+  static dateNotExceed = (sibling: string, months: number): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null => {
+      if (control?.parent) {
+        const siblingControl = control.parent.get(sibling) as CustomFormControl;
+        const siblingDate = new Date(siblingControl.value);
+        siblingDate.setMonth(siblingDate.getMonth() + months);
+
+        if (new Date(control.value) > siblingDate) {
```
Comment (gjulien-bjss) [inline src/app/forms/validators/custom-validators.ts:189]:
```suggestion
        const maxDate = new Date(siblingControl.value);
        maxDate.setMonth(maxDate.getMonth() + months);

        if (new Date(control.value) > maxDate) {
```

Actually now that I think about it doesn't make sense for it to be called `siblingDate` as it's never used in that contect in the code

**Review by mcadenas-bjss — APPROVED** [review]

**Review by mcadenas-bjss — APPROVED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Pull request merged**